crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -302,126 +302,186 @@ async fn sys_validation_workflow_inner(
         }
     }
 
-    let mut warrants = vec![];
-
-    let (mut summary, invalid_ops, forked_pairs) = workspace
-        .dht_db
-        .write_async(move |txn| {
-            let mut summary = OutcomeSummary::default();
-            let mut invalid_ops = vec![];
-            let mut forked_pairs = vec![];
-
-            for (hashed_op, outcome) in validation_outcomes {
-                let (op, op_hash) = hashed_op.into_inner();
-                let op_type = op.get_type();
-
-                if let DhtOp::ChainOp(chain_op) = &op {
-                    // Author a ChainFork warrant if fork is detected
-                    let action = chain_op.action();
-                    if let Some(forked_action) = detect_fork(txn, &action)? {
-                        let signature = chain_op.signature().clone();
-                        let action_hash = action.to_hash();
-                        forked_pairs.push((
-                            action.author().clone(),
-                            ((action_hash, signature), forked_action),
-                        ));
+    #[cfg(not(feature = "unstable-warrants"))]
```

Comment:
I thought of leaving this code of detecing forks and invalid ops and modifying the summary untouched and just put the insertion of warrants into the DB behind the feature. That would be easier to read but more mixed with warranting.